federation/apis/federation/types.go

```diff
@@ -108,3 +108,27 @@ type ClusterList struct {
 	// List of Cluster objects.
 	Items []Cluster `json:"items"`
 }
+
+// Temporary/alpha stuctures to support custom replica assignments within FederatedReplicaSet.
+
+// A set of preferences that can be added to federated version of ReplicaSet as a json-serialized annotation.
+// The preferences allow the user to express in which culsters he wants to put his replicas within the
+// mentiond FederatedReplicaSet.
+type FederatedReplicaSetPreferences struct {
+	// A maping between cluser names and preferences regarding local replicasets in these clusters.
+	// "*" applies to all clusters if explicint mapping is not provided. It is assumed that cluster
+	// should not have any replicas scheduled.
```

Comment:
nit: I don't understand how these two sentences interact.  Do wildcard "*" clusters get replicas?  I assume that the answer is yes, but that's not clear in these two sentences.